minicluster/src/main/java/alluxio/multi/process/MultiProcessCluster.java

```diff
@@ -497,6 +497,16 @@ public synchronized void updateMasterConf(PropertyKey key, @Nullable String valu
     mMasters.forEach(master -> master.updateConf(key, value));
   }
 
+  /**
+   * Updates internal master list with an external address.
```

Comment:
are there any guidelines on when to use this method? This seems somewhat hack-y.

Reply:
It's used if the cluster contains more masters than it itself created.
There is no scenario other than this test to use that method.

Reply:
As I mentioned, `MultiProcessCluster` implementation is too complex to support this scenario without. The use of internal master list is not consistent and it assumes cluster size is fixed when managing the masters, but it assumes variable length when creating clients/contexts/etc.. So that's why I think extending internal master list is not as hacky as it looks, as we are just updating the internal master list with the master created external to this cluster.

Reply:
Ok, that sounds good. Could you maybe add a little bit of context in this javadoc? Like, when/how this method is useful?